src/app/github-link-area/github-link-area.component.ts

```diff
@@ -0,0 +1,116 @@
+import {
+  Component,
+  Input,
+  ViewEncapsulation,
+  OnChanges,
+  SimpleChanges,
+  ViewChild,
+  ElementRef
+} from '@angular/core';
+
+import { GitHubLinkService } from './github-link.service';
+
+/**
+ * GitHub Link Area component. This shows a formatted link to GitHub with an
+ * indicator of whether the GitHub issue is open or closed in an text area.
+ * Works only for public remote repositories for now. See example for how to
+ * use this.
+ */
+@Component({
+  encapsulation: ViewEncapsulation.None,
+  selector: 'github-link-area',
+  styles: [ `
+    .gh-link-open { color: @color-pf-red; }
+    .gh-link-closed { color: @color-pf-green; }
+    .gh-link-error { color: @color-pf-orange; }
+    `],
+  templateUrl: './github-link-area.component.html'
+})
+export class GitHubLinkAreaComponent implements OnChanges {
+
+  @Input('content') content: string;
+
+  constructor(private gitHubLinkService: GitHubLinkService) {}
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes && changes.content) {
+      this.updateLinkTexts();
+      this.updateLinks();
+    }
+  }
+
+  /*
+   * Replaces the match (which should be the default icon) with an icon that
+   * indicates the status contained in linkData.
+   */
+  replaceLink(linkData: any): void {
+    this.content = this.content.split(linkData.match).join(
+      (linkData.state === 'open' ? '<span class="gh-link-open fa fa-clock-o"></span>' : '') +
```

Comment:
color styling should come after the icon definitions: `<span class="fa fa-clock-o gh-link-open"></span>`